components/camel-smb/src/main/java/org/apache/camel/component/smb/converter/SmbConverter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.smb.converter;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.camel.Converter;
+
+@Converter(generateLoader = true)
+public class SmbConverter {
+
+    @Converter
+    public InputStream toInputStream(com.hierynomus.smbj.share.File smbFile) {
+        return smbFile.getInputStream();
+    }
+
+    @Converter
+    public byte[] toByteArray(com.hierynomus.smbj.share.File smbFile) {
```

Comment:
I wonder if there should also be a toString converter that reads all the bytes and adds into a string. The charset can be taken from exchange (see maybe some of the other converters in core)

Reply:
I think there's "risk" here in the sense that the files are not text ones. So, it could make it easy for the users to confuse things. I think the `byte[]` one kinda forces them to deal with that appropriately.

wdyt?

Reply:
ok